hbase/stackable/bin/hbase-entrypoint.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# Entrypoint for HBase that ensures services are shutdown gracefuly
+#
+set -x
+set -euo pipefail
+
+# master, regionserver, rest
+HBASE_ROLE_NAME="$1"
+# k8s service name for this role+group combo
+# <svc-name>.<namespace>.svc.cluster.local
+HBASE_ROLE_SERVICE_NAME="$2"
+# 16010 for master, 16020 for regionservers etc.
+HBASE_ROLE_SERVICE_PORT="$3"
+
+HBASE_ROLE_SERVICE_HOST="${HOSTNAME}.${HBASE_ROLE_SERVICE_NAME}"
+
+REGION_MOVER_OPTS="--regionserverhost ${HBASE_ROLE_SERVICE_HOST}:${HBASE_ROLE_SERVICE_PORT} --operation unload ${REGION_MOVER_OPTS}"
+
+if [ -f /stackable/kerberos/krb5.conf ]; then
+  KERBEROS_REALM=$(grep -oP 'default_realm = \\K.*' /stackable/kerberos/krb5.conf)
+  export KERBEROS_REALM
+fi
+
+prepare_signal_handlers() {
+  unset term_child_pid
+  unset term_kill_needed
+  trap handle_term_signal TERM
+}
+
+handle_term_signal() {
+  if [ "${term_child_pid}" ]; then
+    if [ "regionserver" == "${HBASE_ROLE_NAME}" ]; then
+      echo "Start pre-shutdown"
+      # REGION_MOVER_OPTS is a string that contains multiple arguments and needs to be spliced here
+      # therefore disable shellcheck for this line
+      # shellcheck disable=SC2086
+      /stackable/hbase/bin/hbase org.apache.hadoop.hbase.util.RegionMover ${REGION_MOVER_OPTS}
+      echo "Done pre-shutdown"
+    fi
+    kill -TERM "${term_child_pid}" 2>/dev/null
+  else
+    term_kill_needed='yes'
+  fi
+}
+
+wait_for_termination() {
+  set +e
+  term_child_pid=$1
+  if [[ -v term_kill_needed ]]; then
+    kill -TERM "${term_child_pid}" 2>/dev/null
+  fi
+  wait "${term_child_pid}" 2>/dev/null
+  trap - TERM
```

Comment:
What does this do? Run nothing on SIGTERM?

Reply:
[Copy&pasta from op-rs.](https://github.com/stackabletech/operator-rs/blob/82a0ef6d14c6f4e144ee1b4f5eff6abebf0108f2/crates/stackable-operator/src/utils/bash.rs#L40) 

I don't quite get understand it either.